include/fmt/chrono.h

```diff
@@ -2016,28 +2057,52 @@ struct formatter<std::chrono::time_point<std::chrono::system_clock, Duration>,
     this->do_parse(default_specs.begin(), default_specs.end());
   }
 
-  template <typename FormatContext>
-  auto format(std::chrono::time_point<std::chrono::system_clock> val,
+  template <typename FormatContext, typename Rep, typename Period>
+  auto format(std::chrono::time_point<std::chrono::system_clock,
+                                      std::chrono::duration<Rep, Period>>
+                  val,
               FormatContext& ctx) const -> decltype(ctx.out()) {
-    return formatter<std::tm, Char>::format(localtime(val), ctx);
+    detail::tm_ss tm_ss_val;
+    tm_ss_val.tm =
+        localtime(std::chrono::time_point_cast<std::chrono::seconds>(val));
+
+    constexpr bool reasonable_subsec_available =
+        Period::num == 1 &&
+        (Period::den == 1e1 || Period::den == 1e2 || Period::den == 1e3 ||
+         Period::den == 1e4 || Period::den == 1e5 || Period::den == 1e6 ||
+         Period::den == 1e7 || Period::den == 1e8 || Period::den == 1e9 ||
+         Period::den == 1e10 || Period::den == 1e11 || Period::den == 1e12 ||
+         Period::den == 1e13 || Period::den == 1e14 || Period::den == 1e15);
```

Comment:
Can we support arbitrary fractions and not just 1/power-of-10?

Reply:
Implicitly done.